src/google/protobuf/io/zero_copy_stream_impl_lite.cc

```diff
@@ -150,22 +150,29 @@ int64 ArrayOutputStream::ByteCount() const {
 // ===================================================================
 
 StringOutputStream::StringOutputStream(string* target)
-  : target_(target) {
+  : target_(target), byte_count_((target_ != NULL) ? target_->size() : 0) {
```

Comment:
Trying to emulate the existing behavior. If target is "abc" with size 3, current behavior is to start writing after "abc".